### PR TITLE
Add SSL pinning to EMS requests

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

File: TradeItIosTicketSDK2/SSLPinningDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    private func pathToPinnedServerCertificate() -> String? {
+        let bundle = TradeItBundleProvider.provide()
+        let file = { () -> String? in
+            switch (TradeItSDK.environment) {
+            case TradeItEmsProductionEnv: return "server-prod"
+            case TradeItEmsTestEnv: return "server-qa"
+            default: return nil
+            }
+        }()
+        return bundle.path(forResource: file, ofType: "der")
```
Comment (jsom) [inline TradeItIosTicketSDK2/SSLPinningDelegate.swift]: @mitochondrion is this something worth caching so I'm not looking it up every time?

File: TradeItIosEmsApi/TradeItConnector.m [hunk 2]
```diff
@@ -413,7 +419,10 @@ -(void) sendEMSRequest:(NSMutableURLRequest *)request
     */
 
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^(void){
-        NSURLSession *session = [NSURLSession sharedSession];
+        SSLPinningDelegate *pinningDelegate = [[SSLPinningDelegate alloc] init];
```
Comment (jsom) [inline TradeItIosEmsApi/TradeItConnector.m]: Creating this new every time seems odd. Maybe something to cache as well.

File: TradeItIosTicketSDK2/SSLPinningDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import Foundation
+
+@objc public class SSLPinningDelegate: NSObject, URLSessionDelegate {
+    @objc public func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+        let serverTrust = challenge.protectionSpace.serverTrust
+        let certificate = SecTrustGetCertificateAtIndex(serverTrust!, 0)
+
+        if(!isLocal() && isServerTrusted(challenge, serverTrust: serverTrust) && isSSLCertificateMatching(certificate)) {
```
Comment (mitochondrion) [inline TradeItIosTicketSDK2/SSLPinningDelegate.swift]: @jsom Does local need to be a separate clause?

Also `if(` -> `if (`.

File: TradeItIosEmsApi/TradeItConnector.m (hunk 2 above)
Comment (mitochondrion) [inline TradeItIosEmsApi/TradeItConnector.m]: Make it a static member.

File: TradeItIosTicketSDK2/SSLPinningDelegate.swift (hunk 1 above)
Comment (mitochondrion) [inline TradeItIosTicketSDK2/SSLPinningDelegate.swift]: Yeah, can switch to static. Won't ever change while the app is running.

File: TradeItIosTicketSDK2/SSLPinningDelegate.swift (hunk 3 above)
Comment (jsom) [inline TradeItIosTicketSDK2/SSLPinningDelegate.swift]: Yes it does

**Pull request closed without merging**